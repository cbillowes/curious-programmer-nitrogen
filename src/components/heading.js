import React from 'react'
import styled from 'styled-components'
import Constants from '../../gatsby-data'

const Heading = styled.div`
  font-family: ${Constants.theme.fonts.sans};
  font-size: 130%;
  line-height: 2.75rem;
`

const H1 = ({ className, children }) => {
  return (
    <Heading className={className}><h1>{children}</h1></Heading>
  )
}

const H2 = ({ className, children }) => {
  return (
    <Heading className={className}><h2>{children}</h2></Heading>
  )
}

const H3 = ({ className, children }) => {
  return (
    <Heading className={className}><h3>{children}</h3></Heading>
  )
}

const H4 = ({ className, children }) => {
  return (
    <Heading className={className}><h4>{children}</h4></Heading>
  )
}

export { H1, H2, H3, H4 }
