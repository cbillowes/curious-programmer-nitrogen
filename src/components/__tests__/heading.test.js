import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { componentTranslatesTo } from './helpers'
import { H1, H2, H3, H4 } from '../heading'

Enzyme.configure({ adapter: new Adapter() })

describe('Heading', () => {

  it('should render level one', () => {
    const hasTranslated = componentTranslatesTo(
      <H1>This is a h1</H1>,
      <h1>This is a h1</h1>
    )
    expect(hasTranslated).toBe(true)
  })

  it('should render level two', () => {
    const hasTranslated = componentTranslatesTo(
      <H2>This is a h2</H2>,
      <h2>This is a h2</h2>
    )
    expect(hasTranslated).toBe(true)
  })

  it('should render level three', () => {
    const hasTranslated = componentTranslatesTo(
      <H3>This is a h3</H3>,
      <h3>This is a h3</h3>
    )
    expect(hasTranslated).toBe(true)
  })

  it('should render level four', () => {
    const hasTranslated = componentTranslatesTo(
      <H4>This is a h4</H4>,
      <h4>This is a h4</h4>
    )
    expect(hasTranslated).toBe(true)
  })

})
