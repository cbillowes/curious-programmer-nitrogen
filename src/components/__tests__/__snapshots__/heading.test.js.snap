// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should capture class names 1`] = `
<div>
  <h1
    className="un"
  >
    un
  </h1>
  <h2
    className="deux"
  >
    deux
  </h2>
  <h3
    className="trois"
  >
    trois
  </h3>
  <h4
    className="quatre"
  >
    quatre
  </h4>
</div>
`;

exports[`Heading should not render when empty 1`] = `<div />`;

exports[`Heading should render 1`] = `
<div>
  <h1>
    Don't panic.
  </h1>
  <h2>
    Time is an illusion. Lunchtime doubly so.
  </h2>
  <h3>
    So long and thanks for all the fish.
  </h3>
  <h4>
    The ships hung in the sky in much the same way that bricks don't.
  </h4>
</div>
`;
